pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -417,26 +420,76 @@ func toRecipeConfigTerraformProvidersDatamodel(config *RecipeConfigProperties) m
 	for k, v := range config.Terraform.Providers {
 		dm[k] = []datamodel.ProviderConfigProperties{}
 
-		for _, providerAdditionalProperties := range v {
+		for _, providerConfigProperties := range v {
+			var additionalProperties map[string]any
+			var providerSecrets map[string]datamodel.SecretReference
+
+			// Info: https://json-schema.org/understanding-json-schema/reference/object#additionalproperties
+			additionalProperties = providerConfigProperties.AdditionalProperties
+			providerSecrets = getSecretsFromProviderProperties(providerConfigProperties.Secrets)
+
 			dm[k] = append(dm[k], datamodel.ProviderConfigProperties{
-				AdditionalProperties: providerAdditionalProperties,
+				AdditionalProperties: additionalProperties,
+				Secrets:              providerSecrets,
```

Comment:
We can inline this part too.

```suggestion
				AdditionalProperties: providerConfigProperties.AdditionalProperties,
				Secrets:              getSecretsFromProviderProperties(providerConfigProperties.Secrets),
```